### ISSUE TITLE
request: allow [pack] to initialize symbol values

### ISSUE THREAD

**Issue by porres** [pure-data/pure-data#990]
In pack, "f/float" specify a float type, "p/pointer" specify a pointer type, and "s/symbol" specify a symbol type with a default value of "symbol symbol".

Since number values also set the type to a float and initialize the float to that value, I'd like to also propose that different symbol atoms (other than "f/float/p/pointer/s/symbol") may initialize the symbol value to something else.

**Comment by umlaeute** [comment]
so banging  `[pack f s b]` will output a `[list 0  symbol b(` message?

or to re-phrase my concern at https://github.com/pure-data/pure-data/issues/354#issuecomment-422503530:

`[pack float gloat]` will give you `[list 0 gloat(`?

sidenote: i think the entire idea to use symbols (`s`, `f`, `pointer`,...) to denote the type is somewhat broken in the first place. but it's been there since before some of my students were born. i guess we will never be able to change it.

**Comment by porres** [comment]
> so banging [pack f s b] will output a [list 0 symbol b( message?

yup

> or to re-phrase my concern at #354 (comment):
> 
> [pack float gloat] will give you [list 0 gloat(?

let me see if I get the concern, you fear that the object won't give you a "gloat: bad type" error if you make a typo? I think the error would still be evident when trying the patch, and nothing would break...

**Comment by porres** [comment]
> I think the error would still be evident when trying the patch, and nothing would break...

Oh, I see that the object will default to a "float" type in such cases... this changes things a bit, as so there's a chance someone ignored a pretty clear and findable error and even still saved the patch as such because it was apparently working... and is relying on ignoring the error instead of fixing it... in such cases the behaviour of the patch will change and affect the result, but I think that the old patch was wrong and needed fixing anyway...

**Comment by umlaeute** [comment]
no. i fear that ~~somebody obsessed with making everything consistent and documenting it~~ people will come and ask "why does `[pack f s b]` not give me a `[list f s b(`?" and keep creating issues about that. and all of them will need to be rejected.

because this convenience method will add a major inconsistency that cannot be properly resolved.

**Comment by porres** [comment]
That assumes people usually open issues without reading help files. I don't know about that, but if that's the case, it's hard to predict what they'll be confused about, since they don't read the help file anyway...

Anyway, I agree and was aware about the inconsistency, and a better design could have prevented that, but we know that ship sailed long ago. This would of course be clearly documented and inconsistencies are ok if they're well documented and many of them already are in Pd. From the top of my head, here's an excerpt from the help file of [symbol] as an example: _note: unlike "float", etc., there's no "s" message to forward to another object -- that would conflict with the function of converting arbitrary messages to symbols._

The thing is that eventually I needed to initialize a symbol value for pack and it just presents a headache that could solved by this. I can't think of a better solution.
